packages/lib-classifier/src/store/AnnotatedSteps/AnnotatedStep/AnnotatedStep.js

```diff
@@ -0,0 +1,32 @@
+import { types } from 'mobx-state-tree'
+
+import Step from '@store/WorkflowStepStore/Step'
+import { annotationModels } from '@plugins/tasks'
+
+const TaskAnnotation = types.union(...annotationModels)
+
+const AnnotatedStep = types.model('AnnotatedStep', {
+  id: types.identifier,
+  step: types.safeReference(Step),
+  annotations: types.array(types.safeReference(TaskAnnotation)),
+})
+.views(self => ({
+    /** Check if the step tasks are complete with the current annotations. */
+    get isComplete() {
+      return self.step.isComplete(self.annotations)
+    },
+    /** Get the next step key from any single choice answers, or the current step otherwise. */
+    nextStepKey() {
+      // assume only one branching question per step.
+      const [ singleChoiceAnnotation ] = self.annotations.filter(annotation => annotation.taskType === 'single')
+      if (singleChoiceAnnotation) {
+        const [singleChoiceTask] = self.step.tasks.filter(task => task.taskKey === singleChoiceAnnotation.task)
+        const selectedAnswer = singleChoiceTask?.answers[singleChoiceAnnotation.value]
+        return selectedAnswer?.next
+      }
+      return self.step.next
```

Comment:
Maybe this can also be moved down into the Step model. Something like
```js
return self.step.nextStepKey(annotations)
```

That would make this more consistent with `step.isComplete`, and keeps all the step logic in one place.

Reply:
Done in da32e387e29ca3cf7874d19d3034f019ec1ed931